extensions/ql-vscode/src/vscode-tests/no-workspace/history-item-label-provider.test.ts

```diff
@@ -126,7 +126,35 @@ describe('HistoryItemLabelProvider', () => {
       expect(labelProvider.getShortLabel(fqi)).to.eq('query-name');
     });
 
-    function createMockRemoteQueryInfo(userSpecifiedLabel?: string, repositoryCount?: number) {
+    describe('when results are present', () => {
+      it('should display results if there are any', () => {
+        const fqi = createMockRemoteQueryInfo({ resultCount: 16, repositoryCount: 2 });
+
+        config.format = '%t %q %d %s %f %r %%';
+        expect(labelProvider.getLabel(fqi)).to.eq(`${dateStr} query-name (javascript) 2 repositories in progress query-file.ql (16 results) %`);
+      });
+    });
+
+    describe('when results are not present', () => {
+      it('should skip displaying them', () => {
+        const fqi = createMockRemoteQueryInfo({ resultCount: 0, repositoryCount: 2 });
+
+        config.format = '%t %q %d %s %f %r %%';
+        expect(labelProvider.getLabel(fqi)).to.eq(`${dateStr} query-name (javascript) 2 repositories in progress query-file.ql  %`);
```

Comment:
Hmm, there's a double space at the end here. It shows up in the screenshots too, though it isn't too noticeable unless you're looking for it. I'm wondering how much we care about that. It's better than having the `()` but it's not perfect.

We could try moving the space after the paren into the replacement string and out of the format string. What do you think?

Reply:
I had a look at moving the space into the replacement string. Unfortunately, the spaces are set extension-wide [via `this.config.format`](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/src/history-item-label-provider.ts#L26-L28). 

The config is built from whatever list of elements are passed to it, like we do [here](https://github.com/github/vscode-codeql/blob/2cf5b39cfeaf5306bd8ade30607c62112deb1823/extensions/ql-vscode/src/history-item-label-provider.ts#L81-L87):

```
private getRemoteInterpolateReplacements(item: RemoteQueryHistoryItem): InterpolateReplacements {
    return {
      t: new Date(item.remoteQuery.executionStartTime).toLocaleString(env.language),
      q: `${item.remoteQuery.queryName} (${item.remoteQuery.language})`,
      d: this.buildRepoLabel(item),
      r: `(${pluralize(item.resultCount, 'result', 'results')})`,
      s: item.status,
      f: path.basename(item.remoteQuery.queryFilePath),
      '%': '%'
    };
  }
```

and then builds something like what we define in the tests: 

```
config.format = '%t %q %d %r %s %f  %%';
```

Reply:
We _could_ add some extra handling of spaces during interpolation. After the interpolated string is generated, we can then squash all whitespace into a single space. I don't see a use case for fine-grained control over whitespace. Any repeated whitespace is most likely due to odd interactions with interpolation, like this. So, I think squashing would give a nicer experience.

This is just like HTML does, which makes me think that since vscode is an electron app, do you know if duplicate spaces actually render in the UI?